### ISSUE TITLE
Incorrect handling of default arguments in cdef functions

### ISSUE THREAD

**Issue by antocuni** [cython/cython#1630]
Consider the following code:

```cython
cdef bar(int x=1, int y=2):
    return x + y

cdef foo():
    print bar()
    print bar(x=0)
    print bar(y=0)
```

The first two invocations of `bar()` work correctly; however, the third generates an error:
```
mytest.pyx:7:13: C function call is missing argument 'x'
```

**Comment by mhsmith** [comment]
After experimenting and looking at the generated code, it's clear that although you can specify cdef keyword argument values in any order, you can only omit trailing arguments, like in C++.

This should be documented (or fixed).

**Comment by scoder** [comment]
Yes. PRs for either of the two (docs/feature) would be appreciated.

**Comment by mhsmith** [comment]
A further note: you *can* do this if the function is `cpdef`. Cython uses the fast calling mechanism if possible, and falls back on the Python mechanism if you skipped an argument, though that's probably too internal a detail to document.
